lighthouse-core/runner.js

```diff
@@ -45,54 +44,7 @@ class Runner {
        */
       const lighthouseRunWarnings = [];
 
-      const sentryContext = Sentry.getContext();
-      Sentry.captureBreadcrumb({
-        message: 'Run started',
-        category: 'lifecycle',
-        data: sentryContext && sentryContext.extra,
-      });
-
-      // User can run -G solo, -A solo, or -GA together
-      // -G and -A will run partial lighthouse pipelines,
-      // and -GA will run everything plus save artifacts and lhr to disk.
-
-      // Gather phase
-      // Either load saved artifacts off disk or from the browser
-      let artifacts;
-      let requestedUrl;
-      if (settings.auditMode && !settings.gatherMode) {
-        // No browser required, just load the artifacts from disk.
-        const path = Runner._getDataSavePath(settings);
-        artifacts = assetSaver.loadArtifacts(path);
-        requestedUrl = artifacts.URL.requestedUrl;
-
-        if (!requestedUrl) {
-          throw new Error('Cannot run audit mode on empty URL');
-        }
-        if (runOpts.url && !URL.equalWithExcludedFragments(runOpts.url, requestedUrl)) {
-          throw new Error('Cannot run audit mode on different URL');
-        }
```

Comment:
Removing this check should be the only functional change to legacy mode. I didn't think it was needed because we can proceed with the `requestedUrl` from the artifacts. I'd be open to adding this back as a non-fatal warning or something in a different location.

Reply:
The check makes more sense from the context of the CLI. Currently you can do `-G` and then `-A` separately but you must use the same URL, otherwise it errors because you probably made a mistake. Perhaps instead, we should have `-A` mode only use what is saved in the gathering step–url, settings, etc. and throw if a url is also provided

Reply:
I would prefer just logging a warning, since we could just ignore the passed in URL.

Reply:
i'm fine with just a warning. -G/A is pretty advanced usage so we don't really need these runtime failures to protect us, IMO

Reply:
A report warning, then, not logger. This will otherwise be ignored and is sure to mess me up eventually so a loud warning is necessary :)

Reply:
I would prefer switching to a fatal warning than that. We would need to send the warning message all the way down to `Runner.run` where run warnings are collected. Removing URL checks from `Runner.run` is part of the reason I created this PR.

The warning is quiet, but Lighthouse is still performing the more correct and expected operation IMO.